src/AlphaVSS.Platform/Source/VssAsyncResult.cpp

```diff
@@ -144,5 +150,41 @@ namespace Alphaleonis { namespace Win32 { namespace Vss
 
    void VssAsyncResult::Cancel()
    {
+	   if (m_vssAsync != 0)
+	   {
+		   CheckCom(m_vssAsync->Cancel());
+	   }
+   }
+
+   UInt32 VssAsyncResult::QueryStatus()
+   {
+	   if (m_vssAsync != 0)
+	   {
+		   HRESULT
+			   hr = 0;
+		   Int32
+			   resv = 0;
+		   CheckCom(m_vssAsync->QueryStatus(&hr, &resv));
+		   return hr;
+	   }
+	   return S_OK;
    }
+
+   bool VssAsyncResult::Wait(TimeSpan TimeOut)
+   {
+	   if (m_vssAsync != 0)
+	   {
+		   UInt32
+			   ms = (UInt32)TimeOut.TotalMilliseconds;
+		   if (ms <= 0)
```

Comment:
Why this default value for a negative TimeSpan? Seems we should either wait indefinitely if TimeSpan is `Timeout.InfiniteTimeSpan`, or throw an exception?